src/modules/displaymanager/main.py

```diff
@@ -440,21 +445,36 @@ def run():
             if default_desktop_environment is not None:
                 os.system(
                     "sed -i -e \"s/^.*user-session=.*/user-session={!s}/\" "
-                    "{!s}/etc/lightdm/lightdm.conf".format(
+                    "{!s}".format(
                         default_desktop_environment.desktop_file,
-                        root_mount_point
+                        lightdm_conf_path
                         )
                     )
 
-            if default_desktop_environment.desktop_file == "deepin":
-                os.system(
-                    "sed -i -e \"s/^.greeter-session=.*"
-                    "/greeter-session=lightdm-deepin-greeter/\" "
-                    "{!s}/etc/lightdm/lightdm.conf".format(
-                        root_mount_point
-                        )
+            # configure lightdm-greeter
+            greeter_path = os.path.join(
+                root_mount_point, "usr/share/xgreeters"
+                )
+            if (os.path.exists(greeter_path)):
+                greeter = os.listdir(greeter_path)[0].split('.')[0]
+                greeter_desktop = os.path.join(
+                    greeter_path,
+                    greeter,
+                    ".desktop"
                     )
 
+                if (os.path.exists(greeter_desktop)):
+                    libcalamares.utils.debug("configure {!s}".format(greeter))
+                    os.system(
+                        "sed -i -e "
+                        "\"s/^.*greeter-session=.*/greeter-session={!s}/\" "
+                        "{!s}".format(
+                            greeter,
+                            lightdm_conf_path
+                            )
+                        )
+                else:
+                    return ("No lightdm greeter installed.")
```

Comment:
@philmmanjaro really? Actually this error belongs to condition `if (os.path.exists(greeter_path)):`